### PR TITLE
MAID-2151: Reconnect remote logger if it's disconnected

### PR DESCRIPTION
**Pull request by michaelsproul**

### CODE REVIEW

File: test/src/bin/ws_server.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+extern crate ws;
+
+fn main() {
+    ws::listen("127.0.0.1:55555", |_| {
+        move |msg: ws::Message| {
+            println!("{}", msg.as_text()?);
+            Ok(())
+        }
+    }).unwrap();
```
Comment (ustulation) [inline test/src/bin/ws_server.rs]: we only use `unwrap!()` macro from the unwrap crate

File: test/src/main.rs [hunk 2]
```diff
@@ -28,8 +28,13 @@ use std::time::Duration;
 fn main() {
     unwrap!(safe_log::init(false));
 
+    // FIXME: remove the stuff I added.
```
Comment (ustulation) [inline test/src/main.rs]: Why is this commnent there

**Review by ustulation — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 3]
```diff
@@ -13,7 +13,6 @@
 *.dylib
 
 # Generated by Cargo
-bin/
```
Comment (ustulation) [inline .gitignore:16]: why delete this ?

**Review by ustulation — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (michaelsproul) [reply .gitignore:16]: I think it's out-dated. Cargo doesn't generate files in directories called `bin` any more, and this rule was leading to `test/src/bin/ws_server.rs` being excluded.

File: test/src/main.rs (hunk 2 above)
Comment (michaelsproul) [reply test/src/main.rs]: to force me to write a better test 😁 (and delete this comment before merging)

**Review by ustulation — APPROVED** [review]

**Pull request merged**